content/en/blog/_posts/2023-07-24-mixed-version-proxy-alpha.md

```diff
@@ -0,0 +1,46 @@
+
+## What problem does this solve?
+When a cluster undergoes an upgrade, the kube-apiservers existing at different versions in that scenario can serve different sets (groups, versions, resources) of built-in resources. A resource request made in this scenario may be served by any of the available apiservers, potentially resulting in the request ending up at an apiserver that may not be aware of the requested resource; consequently it being served a 404 not found error which is incorrect. Furthermore, incorrect serving of the 404 errors can lead to serious consequences such as namespace deletion being blocked incorrectly or objects being garbage collected mistakenly.
+
+## How do we solve the problem?
+
+![Alt text](https://github.com/kubernetes/website/assets/26771552/5c0fbb01-5e30-467d-81bf-5739be19b013?raw=true "MVP-flow-diagram")
+
+The new feature “Mixed Version Proxy” provides the kube-apiserver with the capability to proxy a request to a peer kube-apiserver which is aware of the requested resource and hence can serve the request. To do this, a new filter is added to the handler chain in the aggregator.
+
+1. The new filter in the handler chain checks if the request is for a group/version/resource that the apiserver doesn't know about (using the existing [StorageVersion API](https://github.com/kubernetes/kubernetes/blob/release-1.27/pkg/apis/apiserverinternal/types.go#L25-L37)). If so, it proxies the request to one of the apiservers that is listed in the ServerStorageVersion object. If the identified peer apiserver fails to respond (due to reasons like network connectivity, race between the request being received and the controller registering the apiserver-resource info in ServerStorageVersion object), then error 503("Service Unavailable") is served.
+2. To prevent indefinite rerouting of the request, a new header `X-Kubernetes-APIServer-Rerouted:true` is added to the original request once it is determined that the request cannot be served by the original apiserver that received the request and should therefore be proxied. The destination peer apiserver ensures to serve the request locally if it sees this header.
+3. To set the network location of a kube-apiserver that peers will use to proxy requests, the value passed in `--advertise-address` or (when `--advertise-address` is unspecified) the `--bind-address` flag is used. For users with network configurations that would not allow communication between peer kube-apiservers using the addresses specified in these flags, there is an option to pass in the correct peer address as `--peer-advertise-ip` and `--peer-advertise-port` flags that are introduced in this feature.
+
+## How do I enable this feature?
+Following are the required steps to enable the feature:
+
+* Download the latest Kubernetes project (version [v1.28.0](http://v1.28.0/) or later)  
```

Comment:
Doesn't look right. We can link to https://kubernetes.io/releases/download/ BTW

Reply:
Basically I want to emphasize that this feature is available on Kubernetes v1.28, what's a good way to communicate that?

Reply:
I think the only ask here was to not use the `http://v1.28.0/` link. Since it looks like an invalid link accidentally created from the version number.

Reply:
Cool, updated to use https://kubernetes.io/releases/download/